extensions/hibernate-validator/deployment/src/test/java/io/quarkus/hibernate/validator/test/UserCountryNotSetValidatorLocaleTest.java

```diff
@@ -0,0 +1,56 @@
+package io.quarkus.hibernate.validator.test;
+
+import javax.inject.Inject;
+import javax.validation.ValidatorFactory;
+import javax.validation.constraints.Pattern;
+
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.shrinkwrap.api.asset.StringAsset;
+import org.jboss.shrinkwrap.api.spec.JavaArchive;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+import io.quarkus.test.QuarkusUnitTest;
+
+public class UserCountryNotSetValidatorLocaleTest {
+
+    @Inject
+    ValidatorFactory validatorFactory;
+
+    @RegisterExtension
+    static final QuarkusUnitTest test = new QuarkusUnitTest().setArchiveProducer(() -> ShrinkWrap
+            .create(JavaArchive.class).addClasses(Bean.class)
+            .addAsResource(new StringAsset("foo=bar"), "application.properties"))
+            .setBeforeAllCustomizer(new Runnable() {
+                @Override
+                public void run() {
+                    userCountry = System.clearProperty("user.country");
+                    userLanguage = System.clearProperty("user.language");
+                }
+            }).setAfterAllCustomizer(new Runnable() {
+                @Override
+                public void run() {
+                    System.setProperty("user.country", userCountry);
+                    System.setProperty("user.language", userLanguage);
+                }
```

Comment:
Using @BeforeAll and @AfterAll static methods should be the standard way of invoking code before and after tests. Not sure if it's worth adding another strategy to do the same thing, unless I am missing something, of course

Reply:
See my comment above